### PR TITLE
Create exchange_sort.py

### PR DESCRIPTION
**Pull request by bonbon99**
### **Describe your change:**
Added exchange sort as referenced from https://en.wikipedia.org/wiki/Sorting_algorithm#Exchange_sort

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: sorts/exchange_sort.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    [-3, -2, -1]
+    >>> exchange_sort([1, 2, 3, 4, 5])
+    [1, 2, 3, 4, 5]
+    >>> exchange_sort([0, 10, -2, 5, 3])
+    [-2, 0, 3, 5, 10]
+    >>> exchange_sort([])
+    []
+    """
+    for i in range(len(numbers)):
+        for j in range(i + 1, len(numbers)):
```
Comment (L3str4nge) [inline sorts/exchange_sort.py:17]: Calling `len(numbers)` each iteration is inefficient. I would prefer create local variable like `numbers_count = len(numbers)` and use it then in for loops.

**Review by L3str4nge — CHANGES_REQUESTED** [review]
Please also take a look at build fail details.

**Comment by L3str4nge** [comment]
Tests are still failing @bonbon99

**Comment by bonbon99** [comment]
Thank you for reviewing @L3str4nge !

**Review by L3str4nge — APPROVED** [review]

**Pull request merged**